flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2310,7 +2310,12 @@ def test_profile_respected(self):
         mandatory profiles.
         """
         for profile in (c.value for c in MandatoryProfiles.iterconstants()):
-            self.api.create_volume_with_profile(profile)
+            size = GiB(1).to_Byte().value
+            dataset_id = uuid4()
+            self.addCleanup(detach_destroy_volumes, self.api)
+            self.api.create_volume_with_profile(dataset_id=dataset_id,
+                                                size=size,
+                                                profile_name=profile)
 
```

Comment:
This is definitely a more subtle problem that it originally appears. We should probably do a hangout tomorrow to discuss some of the finer points.

Specifically:

- This suggests the API should have a "get profile for this blockdevice_id"
 -- In situations where desired profile can differ from actual profile (due to availability or changing profiles) what should this return?
 -- What should implementations look like for dumb backends like EBS, and smart backends like CIO?
 -- If profiles are more of a suggestion, does that complicate any answers to the former?

Whenever I've wandered down that thought path I've always landed on some things that seem open-ended. I feel if we put it in the interface and it's open ended, we are going to get storage parters implementing multiple interpretations, and that might be a hard place to come out of. So for now, we've sort of been biasing toward, "just put it in EBS and 'know' that its EBS in tests to try out the current API" rather than putting it in the interface. Hopefully later we can ask other driver writers about their experiences and discuss some of the above questions to have a good approach before we extend the interface to be more featureful.

Anyway just something to think about. Hopefully tomorrow we can find some time to do a hangout. It's possible you, Madhuri, or Ryan has more concrete answers to my open questions above and we can make the interface a little larger.